fedimint-core/src/api.rs

```diff
@@ -616,25 +601,23 @@ impl FromStr for WsClientConnectInfo {
         ensure!(variant == Bech32m, "Expected Bech32m encoding");
 
         let bytes: Vec<u8> = Vec::<u8>::from_base32(&data)?;
-        let total_len = bytes.len() as u64;
         let mut cursor = Cursor::new(bytes);
         let mut id_bytes = [0; PK_SIZE];
         cursor.read_exact(&mut id_bytes)?;
 
-        let mut urls = vec![];
-        while cursor.position() < total_len {
-            let mut url_len = [0; 2];
-            cursor.read_exact(&mut url_len)?;
-            let url_len = u16::from_be_bytes(url_len).into();
-            let mut url_bytes = vec![0; url_len];
-            cursor.read_exact(&mut url_bytes)?;
+        let mut url_len = [0; 2];
+        cursor.read_exact(&mut url_len)?;
+        let url_len = u16::from_be_bytes(url_len).into();
+        let mut url_bytes = vec![0; url_len];
+        cursor.read_exact(&mut url_bytes)?;
+        let mut download_token = [0; CONFIG_DOWNLOAD_TOKEN_BYTES];
+        cursor.read_exact(&mut download_token)?;
 
-            let url = std::str::from_utf8(&url_bytes)?;
-            urls.push(url.parse()?);
-        }
+        let url = std::str::from_utf8(&url_bytes)?;
```

Comment:
Why aren't we using `Encodable`/`Decodable` here?

Reply:
This was how it was originally implemented, I guess it would be annoying for decoding to need the `ModuleDecoderRegistry`.

Reply:
We can always create empty `ModuleDecoderRegistry` if we know no modules are going to be involved.